### PR TITLE
test: Add tests for custom system roles

### LINKED ISSUE #5466: Adding dredd tests for custom system roles
**Is your feature request related to a problem? Please describe.**

add dredd tests for custom system roles

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

I'm working on it!

### PR DESCRIPTION
**Pull request by prabhupant**
Fixes #5466 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
adds dredd hooks for custom system roles

#### Changes proposed in this pull request:

- adds tests for custom system roles
- adds factory for custom system roles
- adds factory for panel permissions

### CODE REVIEW

**Comment by prabhupant** [comment]
@iamareebjamal please review

File: tests/hook_main.py [hunk 1]
```diff
@@ -4157,3 +4160,81 @@ def reset_password_patch(transaction):
...
+
+
+@hooks.before("Custom System Roles > Custom System Roles Collection > Create Custom System Role")
+def custom_system_roles_post(transaction):
+    """
+    POST /custom-system-roles
+    :param transaction:
+    :return:
+    """
+    pass
```
Comment (srv-twry) [inline tests/hook_main.py]: why passing this test?

Comment (prabhupant) [reply tests/hook_main.py]: i've added now. please review again

**Comment by prabhupant** [comment]
@srv-twry please review again

**Comment by prabhupant** [comment]
@srv-twry the build has passed. please review again :sweat_smile:

**Pull request merged**